Source/automap.cpp

```diff
@@ -523,8 +523,8 @@ void InitAutomap()
 	memset(automapview, 0, sizeof(automapview));
 
 	for (int y = 0; y < MAXDUNY; y++) {
-		for (int x = 0; x < MAXDUNX; x++)
-			dFlags[x][y] &= ~BFLAG_EXPLORED;
+		for (auto &dFlag : dFlags)
+			dFlag[y] &= ~BFLAG_EXPLORED;

```

Comment:
Not sure about that one, this makes the code less readable I think.

Reply:
I agree

Reply:
Is it this one specifically, because I didn't rename it to &row? Or in general when looping 2d arrays?

Reply:
How about?
```cpp
for (auto &row : dFlags)
    for (auto &dFlag : row)
         dFlag &= ~BFLAG_EXPLORED; 
```

Reply:
yeah either that or the old version, just one auto and one regular loop looks weird

Reply:
Yeah, it's just what clang-tidy was able to do automatically.

Reply:
Well, that just broke everything, anyone has a clue how to rewrite this with ranges in a way that matches the original order.

Reply:
I wouldn't use ranges at all, tbh. Sometimes a good old for loop is still the best and most readable thing to do.

Reply:
Well, we have to figure out how to get clang-tidy to agree with that then.